eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClient.java

```diff
@@ -34,20 +34,21 @@
     private final AtomicBoolean hasConnection = new AtomicBoolean(false);
 
     //TODO (conniey): Can we remove this and replace with an TokenProvider?
-    private final ConnectionStringBuilder connectionStringBuilder;
+    private final ConnectionParameters connectionParameters;
     //TODO (conniey): Replace with configured values in EventHubClientBuilder.
     private final Duration timeout = Duration.ofSeconds(45);
     private final String eventHubName;
 
-    EventHubClient(ConnectionStringBuilder connectionStringBuilder, TokenProvider tokenProvider,
-                   ReactorProvider provider, ReactorHandlerProvider handlerProvider, Scheduler scheduler) {
-        Objects.requireNonNull(connectionStringBuilder, "'connectionStringBuilder' is null");
-        Objects.requireNonNull(connectionStringBuilder.endpoint(), "'connectionStringBuilder.endpoint()' is null.");
 
-        this.connectionStringBuilder = connectionStringBuilder;
-        this.eventHubName = connectionStringBuilder.eventHubName();
-        this.host = connectionStringBuilder.endpoint().getHost();
+    EventHubClient(ConnectionParameters connectionParameters, ReactorProvider provider, ReactorHandlerProvider handlerProvider) {
+        Objects.requireNonNull(connectionParameters, "'connectionParameters' is null");
+        this.connectionParameters = connectionParameters;
+        this.eventHubName = connectionParameters.credentials().eventHubPath();
+        this.host = connectionParameters.credentials().endpoint().getHost();
         this.connectionId = StringUtil.getRandomString("MF");
+
+        TokenProvider tokenProvider = connectionParameters.tokenProvider();
+        Scheduler scheduler = connectionParameters.scheduler();
         this.connectionMono = Mono.fromCallable(() -> ReactorConnection.create(connectionId, host, tokenProvider, provider, handlerProvider, scheduler))
```

Comment:
You don't need to create these two variables (tokenProvider and scheduler) because they aren't used for anything other than passing them to the calling method. In addition, `Mono.fromCallable()` is a deferred call, so the values aren't actually evaluated until someone subscribes to the Mono. This is helpful if there exists a case where the scheduler or tokenProvider can change.

Reply:
Do we still want to have TokenProvider and Scheduler in the ConnectionParamter class, or remove them

Reply:
Keep them. It's fine.